Data source wiz_cloud_configuration_rule causes a panic if a returned rule is not enabled as a control

### Terraform Version and Provider Version
Terraform v1.3.2
terraform-provider-wiz v1.0.6

### Affected Resource(s)
- wiz_cloud_config_rules

### Terraform Configuration Files
```hcl
data "wiz_cloud_config_rules" "test" {
  search = "IAM-078"
}
```

### Debug Output

### Panic Output
```
╷
│ Error: Plugin did not respond
│
│   with wiz_security_framework.test,
│   on main.tf line 174, in resource "wiz_security_framework" "test":
│  174: resource "wiz_security_framework" "test" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with data.wiz_cloud_config_rules.test,
│   on main.tf line 200, in data "wiz_cloud_config_rules" "test":
│  200: data "wiz_cloud_config_rules" "test" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-wiz plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xa7d2a0]

goroutine 42 [running]:
wiz.io/hashicorp/terraform-provider-wiz/internal/provider.flattenCloudConfigurationRules({0xd4b868, 0xc00055cea0}, 0xc0002b4340)
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/internal/provider/data_source_cloud_configuration_rules.go:724 +0x9a0
wiz.io/hashicorp/terraform-provider-wiz/internal/provider.dataSourceWizCloudConfigurationRuleRead({0xd4b868, 0xc00055cea0}, 0x7fde322d3f18, {0xb0d180, 0xc0002f4200})
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/internal/provider/data_source_cloud_configuration_rules.go:690 +0x296a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000122a80, {0xd4b8a0, 0xc0000a1f50}, 0xd, {0xb0d180, 0xc0002f4200})
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000122a80, {0xd4b8a0, 0xc0000a1f50}, 0xc0001b9000, {0xb0d180, 0xc0002f4200})
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000117860, {0xd4b7f8, 0xc00024c880}, 0xc000536ba0)
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:1179 +0x408
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc000335360, {0xd4b8a0, 0xc0004bf9b0}, 0xc000096780)
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:657 +0x45f
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0xc04f60, 0xc000335360}, {0xd4b8a0, 0xc0004bf9b0}, 0xc00055c8a0, 0x0)
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000134700, {0xd597c0, 0xc00048a1a0}, 0xc00016b440, 0xc0003d9c20, 0x124dcf0, 0x0)
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/google.golang.org/grpc/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000134700, {0xd597c0, 0xc00048a1a0}, 0xc00016b440, 0x0)
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/google.golang.org/grpc/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/google.golang.org/grpc/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/t22hi2h/projects/wiz/github/terraform-provider-wiz/vendor/google.golang.org/grpc/server.go:919 +0x294

Error: The terraform-provider-wiz plugin crashed!
```

### Expected Behavior
Returned data source

### Actual Behavior
Panic

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

### References

### Community Note
<!--- Please keep this note for the community --->
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -3,9 +3,6 @@ on:
   pull_request:
     paths-ignore:
       - 'README.md'
-  push:
-    paths-ignore:
-      - 'README.md'
 jobs:
   # ensure the code builds...
   build:

--- a/internal/provider/data_source_cloud_configuration_rules.go
+++ b/internal/provider/data_source_cloud_configuration_rules.go
@@ -721,7 +721,9 @@ func flattenCloudConfigurationRules(ctx context.Context, nodes *[]*vendor.CloudC
 		ruleMap["builtin"] = *b.Builtin
 		ruleMap["opa_policy"] = b.OPAPolicy
 		ruleMap["function_as_control"] = *b.FunctionAsControl
-		ruleMap["control_id"] = b.Control.ID
+		if b.Control != nil {
+			ruleMap["control_id"] = b.Control.ID
+		}
 		ruleMap["graph_id"] = b.GraphID
 		ruleMap["has_auto_remediation"] = *b.HasAutoRemediation
 		ruleMap["remediation_instructions"] = b.RemediationInstructions

